### ISSUE TITLE
`flat/requirements-typescript`: `require-next-type`, `require-throws-type`, `require-yields-type` unexpectedly enabled

### ISSUE THREAD

**Issue by LangLangBart** [gajus/eslint-plugin-jsdoc#1709]
According to the README:

<https://github.com/gajus/eslint-plugin-jsdoc/blob/ebe0d0808725dd90ac866d55b6dec9fd89db6a82/README.md?plain=1#L183-L187>

Using `flat/requirements-typescript` I was surprised that the following rules
are enabled with this config:

- `jsdoc/require-next-type`
- `jsdoc/require-throws-type`
- `jsdoc/require-yields-type`

### Related PR discussion

Discussed if the mentioned type rules should be enabled for the
`flat/recommended-typescript`

- Ref: #1667

## Expected behavior

These rules seem like they should follow the same pattern as
`require-returns-type`, `require-param-type`, `require-property-type`, and moved
to the `*-flavor*` variant only:

- `flat/requirements-typescript-flavor`
- `flat/requirements-typescript-flavor-error`

## Actual behavior

No warning for `@returns` or `@param` without types,  but `@yields`, `@next`,
and `@throws` without types do warn.

### ESLint Config

```json
// package.json
{
  "dependencies": {
    "eslint": "^10.4.0",
    "eslint-plugin-jsdoc": "^63.0.0",
    "typescript": "^6.0.3",
    "typescript-eslint": "^8.59.4"
  },
  "scripts": {
    "lint:check": "eslint"
  },
  "type": "module"
}
```

```js
// eslint.config.js
import parser from '@typescript-eslint/parser'
import jsdoc from 'eslint-plugin-jsdoc'

export default [
  {
    files: ['**/*.ts'],
    languageOptions: { parser },
    plugins: { jsdoc },
    rules: {
      ...jsdoc.configs['flat/requirements-typescript'].rules,
      'jsdoc/require-example': 'off'
    }
  }
]
```

### ESLint sample

```ts
/**
 * @returns A number
 */
function returnsWithoutType() { return 42 }

/**
 * Yields the current count and accepts increments via next.
 * @yields The current count
 * @next The increment value
 */
function* counter(): Generator<number, void, number> {
  let count = 0
  while (true) {
    count += yield count
  }
}

/**
 * @throws An error
 */
function throwsWithoutType() { throw new Error('oops') }

/**
 * @param name A name
 */
function paramWithoutType(name: string) { console.log(name) }
```

```bash
bun run lint:check

# ~/Developer/jsdoc-require-type/test-require-type.ts
#    6:1  warning  @next should have a type    jsdoc/require-next-type
#    6:1  warning  @yields should have a type  jsdoc/require-yields-type
#   18:1  warning  @throws should have a type  jsdoc/require-throws-type

# ✖ 3 problems (0 errors, 3 warnings)
```

## Environment

```bash
bunx envinfo --system --binaries --npmPackages

System:
  OS: macOS 15.7.7
Binaries:
  Node: 26.3.0 - /opt/homebrew/bin/node
  npm: 11.17.0 - /opt/homebrew/bin/npm
  bun: 1.3.14 - /opt/homebrew/bin/bun
  Deno: 2.8.3 - /opt/homebrew/bin/deno
npmPackages:
  eslint: ^10.5.0 => 10.5.0
  eslint-plugin-jsdoc: ^63.0.6 => 63.0.6
  typescript: ^6.0.3 => 6.0.3
  typescript-eslint: ^8.61.1 => 8.61.1
```

**Comment by brettz9** [comment]
Did you see the full discussion in the related PR (which refers to the tsdoc configs to resolve this)? Or is this more of a documentation clarification issue?

**Comment by LangLangBart** [comment]
The PR discussed a different config. I was not interested in the
`flat/recommended-tsdoc` or `flat/recommended-typescript` variant as I only use
**Granular Flat Configs**. 

> Or is this more of a documentation clarification issue?

Right, I was asking myself if the author intentionally enabled those rules for
the `flat/requirements-typescript` config or if it was an oversight, as the
current docs state that only  the `*-flavor*` configs have JSDoc types enabled.

I am seeking clarification, if I should disable those rules myself or if the
`flat/requirements-typescript` config should change.

**Comment by brettz9** [comment]
Ok, re: clarifying documentation, but it seems to me we need to extend what was in the PR to include support for TSDoc-based `flat/requirements-tsdoc` and `flat/requirements-tsdoc-error` granular configs. You could then use that.

**Comment by LangLangBart** [comment]
> `flat/requirements-tsdoc` and `flat/requirements-tsdoc-error` granular configs

It'd work, but at that point I might as well just disable the rules and restore
consistency manually.

With every new config comes a new thing to learn: what does it enable, why does
it exist, what edge cases and pitfalls come with it? An ever-growing list of
configs causes more friction to figure out what is the right config.

I assume you ruled out moving the rules mentioned in the issue title to the
`*-flavor*` variants?

**Comment by brettz9** [comment]
> > `flat/requirements-tsdoc` and `flat/requirements-tsdoc-error` granular configs
> 
> It'd work, but at that point I might as well just disable the rules and restore consistency manually.
> 
> With every new config comes a new thing to learn: what does it enable, why does it exist, what edge cases and pitfalls come with it? An ever-growing list of configs causes more friction to figure out what is the right config.
> 
> I assume you ruled out moving the rules mentioned in the issue title to the `*-flavor*` variants?

The types add semantic value without having a natural TypeScript replacement, so I'm not inclined to remove them from either variant. It is only because of tsdoc that these become problematic.
